### PR TITLE
BREAKING feat(io/streams): Strengthen iterator to readable stream conversion

### LINKED ISSUE #734: Change readableStreamFromAsyncIterator to readableStreamFromIterable
```ts
export function readableStreamFromIterable<T>(
  iterable: Iterable<T> | AsyncIterable<T>,
): ReadableStream<T>;
```
This would be a non-breaking enhancement, if the name change wasn't required.

This helper should be as flexible as possible. The proposed signature would allow passing anything from an array to a generator function iife. Matches `pooledMap()` for instance.

### PR DESCRIPTION
**Pull request by nayeemrmn**
Closes #734.

### CODE REVIEW

**Review by ry — COMMENTED** [review]
I'm in favor of this change - looks clean and useful. 

Are there any objections? This is a breaking API.

**Comment by lucacasonato** [comment]
Looks good to me.

**Comment by caspervonb** [comment]
Seems like a more general purpose improvement to me.

File: io/streams.ts [hunk 1]
```diff
@@ -46,10 +46,24 @@ export function writableStreamFromWriter(
...
+ *        await new Promise(((r) => setTimeout(r, 1000)));
+ *        yield "baz";
+ *      })());
+*/
+export function readableStreamFromIterable<T>(
+  iterable: Iterable<T> | AsyncIterable<T>,
 ): ReadableStream<T> {
+  const iterator: Iterator<T> | AsyncIterator<T> =
+    (iterable as Iterable<T>)[Symbol.iterator]?.() ??
+      (iterable as AsyncIterable<T>)[Symbol.asyncIterator]?.();
```
Comment (nayeemrmn) [inline io/streams.ts:66]: Need to switch these around to match `for await`.

**Review by ry — APPROVED** [review]
LGTM

**Pull request merged**